### ISSUE TITLE
objcopy:firmware.elf: File format not recognized

### ISSUE THREAD

**Issue by jony543** [RudolfGeosits/MSP430-Emulator#13]
Hello,
I compiled my code but the emulator can't recognize the format when loading the file.
Is there instrcution as to which format should I use?

Thanks

**Comment by RudolfGeosits** [comment]
Hi, yes you must use the ELF format, what is the error you are getting? Compiling with CCS or GCC toolchain directly should work but I can help you troubleshoot the issue you're having. Do the example programs work for you?

**Comment by jony543** [comment]
Thanks!
I am using Code Composer studio and the output format is set to ELF. Is this expected to be okay?

Thanks.

**Comment by RudolfGeosits** [comment]
Yes it's expected to be O.K. but there might be a bug. Which version of CCS are you using? What is the error you're experiencing and do the example programs work for you? Feel free to attach your file here so I can analyze the issue. You can also try to compile using GCC directly by:

msp430-gcc -mmcu=msp430g2553 YourProgram.c -o YourProgram.elf

**Comment by jony543** [comment]
I changed the compiler from TI to GNU and now the firmware is loaded, but when running both sample projects (hello world, blink) from code studio.

**Comment by RudolfGeosits** [comment]
Okay, I have an idea of what is going wrong; can you share the file you are trying to load and I can attempt to fix the problem.

**Comment by jony543** [comment]
Thanks. Atached here.
[HelloWorld.zip](https://github.com/RudolfGeosits/MSP430-Emulator/files/4742984/HelloWorld.zip)

**Comment by aLexBeGGy16** [comment]
Were you able to find a fix for this @RudolfGeosits ? Because I'm having the same problem (*INVALID INSTRUCTION*)

**Comment by laksh225** [comment]
Could you share the .c file as that might helpful for anyone debugging the issue.
